app/src/main/java/com/example/campuscollab/service/UserService.java

```diff
@@ -122,12 +145,50 @@ private User mapDocToUser(DocumentSnapshot documentSnapshot) {
         user.setDescription((String) documentSnapshot.get("description"));
         user.setLinkedInUrl((String) documentSnapshot.get("linkedInUrl"));
         user.setMajor((String) documentSnapshot.get("major"));
-        user.setPictureUrl((String) documentSnapshot.get("pictureUrl"));
+        user.setImagePath((String) documentSnapshot.get("imagePath"));
         user.setSkills((List<String>) documentSnapshot.get("skills"));
         user.setResumeUrl((String) documentSnapshot.get("resumeUrl"));
         user.setCreatedDate((Timestamp) documentSnapshot.get("createdDate"));
+        user.setSchool((String) documentSnapshot.get("school"));
 
         return user;
     }
 
+    public AsyncTask<Void, Void, byte[]> getImageBytes(String imagePath) {
+        AsyncTask<Void, Void, byte[]> task = new AsyncTask<Void, Void, byte[]>() {
+            @Override
+            protected byte[] doInBackground(Void... voids) {
+                try{
+                    Task<byte[]> getImageTask = userRepository.getImageBytes(imagePath);
+                    byte[] imageBytes = Tasks.await(getImageTask);
+
+                    return imageBytes;
+                } catch (InterruptedException | ExecutionException | NullPointerException e) {
+                    return null;
+                }
+            }
+        };
+
+        return task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    public AsyncTask<Void, Void, Void> uploadImageBytes(String imagePath, byte[] imageBytes) {
+        AsyncTask<Void, Void, Void> task = new AsyncTask<Void, Void, Void>() {
+            @Override
+            protected Void doInBackground(Void... voids) {
+                try{
+                    UploadTask getImageTask = userRepository.uploadImageBytes(imagePath, imageBytes);
+                    Tasks.await(getImageTask);
+
+                    currentUser.setImagePath(imagePath);
+                    updateUser(currentUser);

```

Comment:
I'm thinking since the images will be displayed pretty small we can probably do some sort of compression here to speed up uploads and retrievals. I'm working on speeding up the backend a bit so I'll look into some compression next week

Reply:
Disregard. Looks like these are Kai's changes got merged that somehow these are carrying over into your pr. Sorry!